src/back/spv/writer.rs

```diff
@@ -1666,8 +1733,11 @@ impl Writer {
             }
         }
 
-        function.blocks.push(block);
-        Ok(id)
+        if branch_id.is_some() {
+            block.termination = Some(super::instructions::instruction_branch(branch_id.unwrap()));
```

Comment:
note: we don't want to do this if there already a `termination` assigned, since it could be `return`

Reply:
Yes true, I think we don't even need to check here if it `is_some`. As it is required to have a termination. So it either has to have a termination already from the statements, otherwise via the `merge_id` parameter.

```rust
  if block.termination.is_none() {
    block.termination = Some(super::instructions::instruction_branch(branch_id.unwrap()));
  }
```